app/views/unattended/provisioning_templates/user_data/userdata_open_vm_tools.erb

```
@@ -0,0 +1,36 @@
+<%#
+kind: user_data
+name: UserData open-vm-tools
+model: ProvisioningTemplate
+oses:
+- CentOS
+- Fedora
+- Debian
+- Ubuntu
+-%>
+# Template for VMWare customization via open-vm-tools
+
+identity:
+  LinuxPrep:
+    domain: <%= @host.domain %>
+    hostName: <%= @host.shortname %>
+    hwClockUTC: true
+    timeZone: <%= host_param('time-zone') || 'UTC' %>
+
+globalIPSettings:
+  dnsSuffixList: [<%= @host.domain %>]
+  <%- @host.interfaces.each do |interface| -%>
+  <%- next unless interface.subnet -%>
+  dnsServerList: [<%= interface.subnet.dns_primary %>, <%= interface.subnet.dns_secondary %>]
+  <%- end -%>
+
+nicSettingMap:
+<%- @host.interfaces.each do |interface| -%>
+<%- next unless interface.subnet -%>
+  - adapter:
+      dnsDomain: <%= interface.domain %>
+      dnsServerList: [<%= interface.subnet.dns_primary %>, <%= interface.subnet.dns_secondary %>]
+      gateway: [<%= interface.subnet.gateway %>]
+      ip: <%= interface.ip %>
+      subnetMask: <%= interface.subnet.mask %>
+<%- end -%>
```

It looks like this is doable, all root elements from this template are not clashing with the rest in the other one.

On the other hand, I assumed that open-vm client will silently ignore elements it does not want. This is typical way of parsing YAML, but that might not be the case. Then we still need two separate templates.

---

Merging the templates might cause even more confusion. I'd prefer to avoid this.

---

That's fair point.